### ISSUE TITLE
Wildcarded declarations

### ISSUE THREAD

**Issue by ludwikbukowski** [amalbuquerque/interceptor#5]
What do you think about the idea of extending the interceptions to support wildcarded declarations like:
```
defmodule Interception.Config do
  use Interceptor.Configurator

  intercept "Intercepted.*/1",
    before: "MyInterceptor.wildcarded_before/1",
    after: "MyInterceptor.widcarded_after/2"
```
  with following callbacks' implementation: 
```
defmodule MyInterceptor do
  def wildcarded_before(mfa),
    do: IO.puts "Intercepted #{inspect(mfa)} before it started."

  def wildcarded_after(mfa, result),
    do: IO.puts "Intercepted #{inspect(mfa)} after it completed. Its result: #{inspect(result)}"
end
```
or even with arity designators: 
```
...
 intercept "Intercepted.*/*",

```
Im just thinking if there could be any problems witch such approach...
Would you be interested in any contribution supporting such feature?

**Comment by amalbuquerque** [comment]
Hi there,

First of all, thanks for the suggestion and for opening the library first issue :)

I like your suggestion, since it allows us to keep the interception configuration module more streamlined.

I don't think there should be any problem with that approach, the only issue I see is that we should also support the wildcard approach with the tuple-based interception config. In this case, the following streamlined configuration using wildcards:

```elixir
defmodule Interception.Config do
  use Interceptor.Configurator

  intercept "Intercepted.*/1",
    before: "MyInterceptor.wildcarded_before/1",
    after: "MyInterceptor.widcarded_after/2"
```

Would be equivalent to this:

```elixir
defmodule Interception.Config do
  def get_intercept_config, do: %{
    {Intercepted, :*, 1} => [
      before: {MyInterceptor, :wildcarded_before, 1},
      after: {MyInterceptor, :widcarded_after, 2}
    ]
  }
end
```

And both configurations should work seamlessly.

To implement this, I think we should ensure that:

- The `Interceptor.Configurator` module allows to pass the intercepted function as `Intercepted.*/*` and will convert this to `{Intercepted, :*, :*}`, and;
- When the `Interceptor` changes the function definitions, it knows that it should intercept the functions `Intercepted.abc/1`, `Intercepted.xyz/2` and `Intercepted.ghi/3` if there is an interception configuration for `Intercepted.*/*`.

If you want to develop this, please do, contributions are more than welcome :raised_hands:. Please let me know if you have any questions.

**Comment by amalbuquerque** [comment]
Resolved with release 0.5.3
